src/borg/testsuite/archiver/debug_cmds.py

```diff
@@ -63,6 +64,64 @@ def test_debug_put_get_delete_obj(self):
         output = self.cmd(f"--repo={self.repository_location}", "debug", "delete-obj", "invalid")
         assert "is invalid" in output
 
+    def test_debug_id_hash_format_put_get_parse_obj(self):
+        """Test format-obj and parse-obj commands"""
+
+        self.cmd(f"--repo={self.repository_location}", "rcreate", RK_ENCRYPTION)
+        data = b"some data"
+        meta_dict = {"some": "property"}
+        meta = json.dumps(meta_dict, ensure_ascii=False).encode()
+
+        self.create_regular_file("plain.bin", contents=data)
+        self.create_regular_file("meta.json", contents=meta)
+
+        output = self.cmd(f"--repo={self.repository_location}", "debug", "id-hash", "input/plain.bin")
+        id_hash = output.strip()
+
+        output = self.cmd(
+            f"--repo={self.repository_location}",
+            "debug",
+            "format-obj",
+            id_hash,
+            "input/plain.bin",
+            "input/meta.json",
+            "output/data.bin",
+            "--compression=zstd,2",
+        )
+
+        output = self.cmd(f"--repo={self.repository_location}", "debug", "put-obj", id_hash, "output/data.bin")
+        assert id_hash in output
+
+        output = self.cmd(f"--repo={self.repository_location}", "debug", "get-obj", id_hash, "output/compressed.bin")
+        assert id_hash in output
+
+        output = self.cmd(
+            f"--repo={self.repository_location}",
+            "debug",
+            "parse-obj",
+            id_hash,
+            "output/compressed.bin",
+            "output/plain.bin",
+            "output/meta.json",
+        )
+
+        with open("output/plain.bin", "rb") as f:
+            data_read = f.read()
+        assert data == data_read
+
+        with open("output/meta.json") as f:
+            meta_read = json.load(f)
+        for key, value in meta_dict.items():
+            assert meta_read.get(key) == value
+
+        assert meta_read.get("size") == len(data_read)
+
+        c = Compressor(name="zstd", level=2)
+        _, data_compressed = c.compress(meta_dict, data=data)
+        assert meta_read.get("csize") == len(data_compressed)
+        assert "ctype" in meta_read
+        assert "clevel" in meta_read
```

Comment:
you can get the type and level from `c` so you can do an assertion of the value, not just of presence.

Reply:
I had to add some repeated sequence so that the data compresses, otherwise because of the heuristic it is falling back to CNONE(i.e. No compression) and the test is failing. Is this alright or should I make sure the test work with incompressible data?

Reply:
Yeah, one needs a good amount of compressible data so it does not fall back to NONE compression (because that has smaller overhead).